### ISSUE TITLE
Define Empty okta_group

### ISSUE THREAD

**Issue by exitcode0** [okta/terraform-provider-okta#1072]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

with the `users` attribute being deprecated on the `okta_group` resource I don't believe there is currently a way with the provider to authoritatively define an empty Okta group
```hcl
resource "okta_group_memberships" "testing_group" {
  group_id = okta_group.testing_group.id
  users = []
}
```
will throw `group (<ID>) did not have expected user memberships after multiple checks`
`okta_group_membership` requires the definition of a user ID

`okta_group_memberships` throwing error when a user isn't defined has been useful to me in some circumstances as I have a python script that imports existing applications and this has allowed me to determine when certain app owner groups have no current members.

I'm now finding myself with a use-case for a group that is emptied on each TF run. mainly for my IT support team granting temp access i.e lost yubikey and want to allow temp TOTP or SMS MFA or temp application access

### New or Affected Resource(s)

- okta_group
- okta_group_membership
- okta_group_memberships

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

- #0000

**Comment by monde** [comment]
Thanks @exitcode0 , it also seem buggy, I'll look into this.

**Comment by exitcode0** [comment]
the issue seems to only crop up when creating a `group_memberships` resource with an empty `users` attr when the group has no users
I believe this is due to [these lines](https://github.com/okta/terraform-provider-okta/blob/51633d5d2fac4ce9fb1be22567739e7061f5b6b7/okta/resource_okta_group_memberships.go#L126-L128)

I've added some commits to my open PR for work in this file at #924

**Comment by monde** [comment]
Released as [v3.28.0](https://github.com/okta/terraform-provider-okta/releases/tag/v3.28.0).
